config/aaa.py

```diff
@@ -240,7 +274,20 @@ def passkey(ctx, secret):
     if ctx.obj == 'default':
         del_table_key('TACPLUS', 'global', 'passkey')
     elif secret:
-        add_table_kv('TACPLUS', 'global', 'passkey', secret)
+        if len(secret) > TACACS_PASSKEY_MAX_LEN:
+            click.echo('Maximum of %d chars can be configured' % TACACS_PASSKEY_MAX_LEN)
+            return
+        elif not is_secret(secret):
+            click.echo(VALID_CHARS_MSG)
+            return
+        config_db = ConfigDBConnector()
+        config_db.connect()
+        outsecret, errs = encrypt_passkey(secret)
+        if not errs: 
+            add_table_kv('TACPLUS', 'global', 'passkey', outsecret)
```

Comment:
Do you test if this binary will break /etc/sonic/config_db.json?

that file generated by sonic-cfggen, and when device reboot content in redis will save/load from it.

Because binary may contain any characters, I suggest convert binary to base64 string.

Reply:
HI Liuh, the hash returned by encrypt_passkey() API is in string format only. Thus there are no issues while adding it into config_db. Please refer to the below snippet.

<SNIP>
>>> out, err = encrypt_passkey("TEST")
>>> print(out)
U2FsdGVkX19yDSbDSlgRXhtBzTThlMFo8YmvfioH/qo=
>>> print(type(out))
<class 'str'>
<SNIP>

Reply:
Thanks, this the result is a base64 format string.
So as my comments in the HLD PR, the yang model need change for longer key value, as you can see the original pass key is 'TEST' and encrypted key almost 10 times longer than original key.

Reply:
@liuh-80 one question related to sonic-system-tacacs.yang model. Is it being used for pushing the TACACS configs or pulling the telemetry data related to TACACS or both?

And in case of increasing the length of passkey leaf, do I need to make any other backward compatibility changes or nothing?